addon/ng2/blueprints/component/index.js

```diff
@@ -16,6 +27,11 @@ module.exports = {
     var parsedPath = dynamicPathParser(this.project, entityName);
 
     this.dynamicPath = parsedPath;
+    
+    if (!validateName(parsedPath.name)) {
+      throw 'Invalid name. Names must be camelCase or kebab-case';
```

Comment:
FWIW, the message might be confusing with single-word names, since `toast` is technically valid camelCase and kebab-case.

More generally, `validateName()` isn't strict at all, which means that all sorts of non-camelCase-or-kebab-case names would pass. (But maybe that's intentional.)